packages/contracts/contracts/TransactionManager.sol

```diff
@@ -110,13 +110,9 @@ contract TransactionManager is ReentrancyGuard, ProposedOwnable, ITransactionMan
     uint256 chain = chainId;
     if (chain == 0) {
       // If not provided, pull from block
-      assembly {
-        _chainId := chainid()
-      }
-    } else {
-      // Use provided override
-      _chainId = chain;
+      chain = block.chainid;
```

Comment:
if chain == 0 couldn't that also be mainnet eth? sry not familiar with api of where this is pulled from

Reply:
no chain == 0 means its uninitiated via the constructor, and so should be taken from the block. the reason we have the option to override the chain is because some chains don't have unique chain ids

Reply:
some chains don't have unique chain ids? woah I did not know that

also, what would eth be then, if not chainid 0?